### PR TITLE
add builtin time period diff func

### PR DESCRIPTION
**Pull request by tangdi**
add period diff func impl in expression

### CODE REVIEW

**Comment by shenli** [comment]
@tangdi Thanks for your PR!
CI is failed. Please make sure you could pass `make dev`.

**Comment by tangdi** [comment]
@shenli  now CI passed. Thanks.

File: expression/builtin_time.go [hunk 1]
```diff
@@ -1975,7 +1966,52 @@ type builtinPeriodDiffSig struct {
...
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_period-diff
 func (b *builtinPeriodDiffSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("PERIOD_DIFF")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	if args[0].IsNull() || args[1].IsNull() {
+		return d, errors.Trace(err)
```
Comment (shenli) [inline expression/builtin_time.go]: return d, nil is more clear.

File: expression/builtin_time_test.go [hunk 2]
```diff
@@ -1426,3 +1426,43 @@ func (s *testEvaluatorSuite) TestSecToTime(c *C) {
 		c.Assert(result, Equals, test.expect)
 	}
 }
+
+func (s *testEvaluatorSuite) TestPeriodDiff(c *C) {
+	tests := []struct {
```
Comment (shenli) [inline expression/builtin_time_test.go]: Add a test case that Period1 or Period2 is null.

**Comment by hanfei1991** [comment]
LGTM

**Comment by hanfei1991** [comment]
@tangdi resolve conficts please.

**Comment by tangdi** [comment]
@hanfei1991  done

**Pull request merged**